### ISSUE TITLE
rule 'prefer-destructuring' and objects

### ISSUE THREAD

**Issue by boggddan** [eslint/eslint#8923]
Hello, could tell me please.

The property of the object is assigned from the property of another object
The EsLint displays an error on this code.

```javascript
/*eslint 'prefer-destructuring': [ 'error',
 { array: true, object: true }, { enforceForRenamedProperties: true } ],
*/

splendingdate.dataset.oldValue = splendingdate.value;
```

Error from the command line:

```bash
  5:1  error  Use object destructuring  prefer-destructuring

✖ 1 problem (1 error, 0 warnings)
```

My file settings `.eslintrc.js`

```javascript
module.exports = {
  env: {
    browser: true,
    es6: true
  },
  extends: '',
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module'
  },
  rules: {
  }
};
```

Help me, to write correct code. Thanks.

**Comment by not-an-aardvark** [comment]
```js
({value: splendingdate.dataset.oldValue} = splendingdate);
```

**Comment by boggddan** [comment]
Thanks, very simple! :+1:

**Comment by Zamiell** [comment]
@not-an-aardvark Can you please add this tip to the documentation [located here](https://github.com/eslint/eslint/blob/master/docs/rules/prefer-destructuring.md)? I was pulling my hair out over this and it took me a while of searching before I stumbled upon this specific issue.
